core/src/main/scala/org/bitcoins/core/crypto/BIP39Seed.scala

```diff
@@ -0,0 +1,65 @@
+package org.bitcoins.core.crypto
+
+import javax.crypto.SecretKeyFactory
+import javax.crypto.spec.PBEKeySpec
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.util.Factory
+import scodec.bits.ByteVector
+
+sealed abstract class BIP39Seed extends NetworkElement {
+  private val MIN_SEED_LENGTH_BYTES = 16
+
+  private val MAX_SEED_LENGTH_BYTES = 64
+  require(
```

Comment:
Please put the invariant as the first thing inside of the class, easier to see for a reader.